OpenRA.Game/Graphics/HardwarePalette.cs

```diff
@@ -52,14 +51,21 @@ public int GetPaletteIndex(string name)
 
 		public void AddPalette(string name, ImmutablePalette p, bool allowModifiers)
 		{
-			if (palettes.Count >= MaxPalettes)
-				throw new InvalidOperationException("Limit of {0} palettes reached. Cannot add {1}.".F(MaxPalettes, name));
 			if (palettes.ContainsKey(name))
 				throw new InvalidOperationException("Palette {0} has already been defined".F(name));
 
 			int index = palettes.Count;
 			indices.Add(name, index);
 			palettes.Add(name, p);
+
+			if (palettes.Count > Height)
+			{
+				Height = Exts.NextPowerOf2(palettes.Count);
+				var oldBuffer = buffer;
```

Comment:
This could be reduced to `Array.Resize(ref buffer, Height * Palette.Size * 4);` for brevity.